pkg/provider/cloud/vsphere/provider.go

```diff
@@ -233,6 +235,16 @@ func (v *VSphere) CleanUpCloudProvider(ctx context.Context, cluster *kubermaticv
 		}
 	}
 
+	// remove orphaned Category finalizer
+	if kuberneteshelper.HasFinalizer(cluster, tagCategoryCleanupFinalizer) {
+		cluster, err = update(ctx, cluster.Name, func(cluster *kubermaticv1.Cluster) {
+			kuberneteshelper.RemoveFinalizer(cluster, tagCategoryCleanupFinalizer)
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I think this could use `kuberneteshelper.TryRemoveFinalizer` instead.

Reply:
I do not think its possible as I do not have a client available in the function, only the `ClusterUpdater`:
`type ClusterUpdater func(context.Context, string, func(*kubermaticv1.Cluster)) (*kubermaticv1.Cluster, error)`

Reply:
True, good point. I take that back :D